### PR TITLE
Swap Functionality

### PR DESCRIPTION
**Pull request by poldo-iov**
Add the swap functionality to the rsk swap ui.

Stilll testing and needs to make sure everything is working

### CODE REVIEW

File: src/components/swap/SwapBox.vue
Comment (coolwather) [inline src/components/swap/SwapBox.vue:380]: This JSON could be in a separated file, to be reusable and easy to change if necessary.

File: src/components/swap/SwapBox.vue
Comment (coolwather) [inline src/components/swap/SwapBox.vue:362]: Bridge was removed from the swap, now the swap and verifying contract is Relayer

File: src/components/swap/SwapBox.vue
Comment (coolwather) [inline src/components/swap/SwapBox.vue:316]: Remove BRIDGE, you can take a look on this script to do the swap.
https://github.com/rsksmart/tokenbridge-backend/blob/main/api/scripts/scriptSwapRwbtc.ts

File: src/components/swap/SwapBox.vue
Comment (poldo-iov) [reply src/components/swap/SwapBox.vue:353]: replaced to process.env.VUE_APP_RELAYER_ADDRESS

File: src/components/swap/SwapBox.vue
Comment (poldo-iov) [reply src/components/swap/SwapBox.vue]: moved to constants

File: src/components/swap/SwapBox.vue
Comment (poldo-iov) [reply src/components/swap/SwapBox.vue]: removed since we dont use it at all, thanks!

File: src/components/swap/SwapBox.vue
Comment (aguschirico) [inline src/components/swap/SwapBox.vue:145]: don't make a hardcoded 18 decimals as default, just leave it to give an error.

File: src/components/swap/SwapBox.vue
Comment (aguschirico) [inline src/components/swap/SwapBox.vue:156]: allowance doesn't apply anymore, i removed it, please check that you merged correctly.

File: src/components/swap/SwapBox.vue
Comment (aguschirico) [inline src/components/swap/SwapBox.vue:217]: this shouln't be here, we don't have approvals anymore.

File: src/components/swap/SwapBox.vue
Comment (aguschirico) [inline src/components/swap/SwapBox.vue:342]: this needs to be adapted to the latest version of the typed data (check tokenbridge-backend script)

File: src/components/swap/SwapBox.vue
Comment (aguschirico) [inline src/components/swap/SwapBox.vue:357]: primary type is no longer Claim, it's Transfer (check latest of tokenbridge-backend)

File: src/constants/message.ts
Comment (aguschirico) [inline src/constants/message.ts:8]: This also needs to match the above mentioned structure.

**Review by aguschirico — CHANGES_REQUESTED** [review]

File: src/components/swap/SwapBox.vue
Comment (poldo-iov) [reply src/components/swap/SwapBox.vue:145]: removed and add a check to throw error if no decimals present

File: src/components/swap/SwapBox.vue
Comment (poldo-iov) [reply src/components/swap/SwapBox.vue:156]: removed, sorry, bad refactor

File: src/components/swap/SwapBox.vue
Comment (poldo-iov) [reply src/components/swap/SwapBox.vue:217]: removed too

File: src/components/swap/SwapBox.vue
Comment (poldo-iov) [reply src/components/swap/SwapBox.vue:345]: but on the script I see that it uses the relayer 
image

File: src/components/swap/SwapBox.vue
Comment (aguschirico) [inline src/components/swap/SwapBox.vue:155]: Just a note here: We are assuming that the ERC20 has 18 decimals here, implicitly (using fromWei). This will work because the WBTC token in RSK will have 18 decimals, but ideally we shouldn't do this. We should ask the contract for the decimals (check the code of the Anyswap ERC20). Don't change it but keep it in mind.

File: src/components/swap/SwapBox.vue
Comment (aguschirico) [inline src/components/swap/SwapBox.vue:214]: if you rebase from main, you will be able to do 
`throw new UserNotificationError(..., ..., ...)`
and that will automatically show the error toast.

File: src/components/swap/SwapBox.vue
Comment (aguschirico) [reply src/components/swap/SwapBox.vue]: then the script is wrong. "Don't trust, verify.":
https://github.com/anyswap/chaindata/blob/3d4640fdfde46e4c34eed84efbb06ab5ea69e88d/AnyswapV6ERC20.sol#L287

Comment (aguschirico) [reply src/components/swap/SwapBox.vue]: @coolwather please review this

**Review by aguschirico — APPROVED** [review]

**Pull request merged**
